velox/exec/HashTable.cpp

```diff
@@ -1603,13 +1644,42 @@ void HashTable<ignoreNullKeys>::eraseWithHashes(
           0,
           [&](const char* group, int32_t row) { return rows[row] == group; },
           [&](int32_t /*index*/, int32_t /*row*/) { return nullptr; },
+          numTombstones_,
           false);
     }
   }
   numDistinct_ -= numRows;
   rows_->eraseRows(rows);
 }
 
+template <bool ignoreNullKeys>
+void HashTable<ignoreNullKeys>::testingSanityCheck() const {
+  VELOX_CHECK_GT(size_, numDistinct_);
```

Comment:
Does the size_ must be greater the numDistinct_ in array mode?

Reply:
Good point. Added.